client/signup/steps/domains/index.jsx

```diff
@@ -744,6 +747,7 @@ class DomainsStep extends React.Component {
 				fallbackSubHeaderText={ fallbackSubHeaderText }
 				stepContent={
 					<div>
+						{ ! this.props.hasAnonId && <QueryExperiments /> }
```

Comment:
Can you please explain what this does? Would we also need to apply a similar check before `<QueryExperiments /` in the plans step?

Reply:
@aneeshd16 The user is supposed to be signed in when they are in the domains step so the `anonId` should not exist because only anonymous can have it. The `<QueryExperiments />` is expected to query the current user's experiments. But the reality is that the `anonId` exists for a while after the user gets the domains step. The `<QueryExperiments />` queries an anonymous' experiments and won't queries again within the time to live, which causes a bug. This only happens in the domains step where the newly signed up user arrives first.

Reply:
> The user is supposed to be signed in when they are in the domains step so the anonId should not exist because only anonymous can have it.

Logged in users can have an anon id and it shouldn't be an issue if they do. What's the bug you're seeing if the user has one?

Reply:
> Logged in users can have an anon id and it shouldn't be an issue if they do.

@withinboredom If that is true, it might be a timing issue. The experiment passed to the custom segments check always has an empty `user_id` when I use `<QueryExperiments />` without `hasAnonId` check.

Reply:
>  it might be a timing issue.

You're right! It is a timing issue. That was a pita to fix in the experiment component...